examples/fluids/problems/blasius.c

```diff
@@ -12,6 +12,41 @@
 #include "../qfunctions/blasius.h"
 #include "stg_shur14.h"
 
+PetscErrorCode getYNodeLocs(const MPI_Comm comm,
+                            const char path[PETSC_MAX_PATH_LEN], PetscReal **pnode_locs,
+                            PetscInt *num_node_locs) {
+  PetscErrorCode ierr;
+  PetscInt ndims, dims[2];
+  FILE *fp;
+  const PetscInt char_array_len = 512;
+  char line[char_array_len];
+  char **array;
+  PetscReal *node_locs;
+  PetscFunctionBeginUser;
+
+  ierr = PetscFOpen(comm, path, "r", &fp); CHKERRQ(ierr);
+  ierr = PetscSynchronizedFGets(comm, fp, char_array_len, line); CHKERRQ(ierr);
+  ierr = PetscStrToArray(line, ' ', &ndims, &array); CHKERRQ(ierr);
+
+  for (PetscInt i=0; i<ndims; i++)  dims[i] = atoi(array[i]);
```

Comment:
Maybe cleaner (and faster, but not important because these files are small) alternative is to use https://petsc.org/main/docs/manualpages/Viewer/PetscViewerASCIIRead/ and broadcast the result.

Reply:
I'm all for giving it a shot, but I'm not sure how to do it. Documentation (that I can find) doesn't describe the format that the viewer would expect, how to handle multiple columns, etc. Grepping through the examples doesn't show much either. Do you have an example I could reference?

Reply:
Maybe check `PetscViewerRead` calls in `plexgmsh.c`, which reads single and multi-column data (ASCII or Binary), a mix of integers and floating point.